src/libs/hooks/use-timer.ts

```diff
@@ -0,0 +1,85 @@
+import {
+	ReadonlySignal,
+	Signal,
+	useComputed,
+	useSignal,
+	useSignalEffect,
+} from "@preact/signals-react";
+import { useCallback } from "react";
+
+type Options = {
+	min: number;
+	sec: number;
+};
+
+type ReturnFunctionType = {
+	resetTimer: () => void;
+	startTimer: () => void;
+	stopTimer: () => void;
+	timeRemaining: ReadonlySignal<string>;
+};
+
+const useTimer = ({ min, sec }: Options): ReturnFunctionType => {
```

Comment:
comment needed what does this function do

Reply:
hook for timer, that calculates time before code expires